### PR TITLE
rebrand Sumokoin to Ryo

### PR DESCRIPTION
**Pull request by psychocrypt**
- add licence to changed files
- rename Sumokoin to Ryo
- update file names and binary names

~~This is not the final daemon the seed nodes and checkpoint nodes must be updated (separate PR).~~

### CODE REVIEW

File: src/cryptonote_config.h [hunk 1]
```diff
@@ -149,7 +150,7 @@ namespace config
   uint64_t const BASE_REWARD_CLAMP_THRESHOLD = ((uint64_t)100000); // pow(10, 5)
   std::string const P2P_REMOTE_DEBUG_TRUSTED_PUB_KEY = "0000000000000000000000000000000000000000000000000000000000000000";
 
-  uint64_t const CRYPTONOTE_PUBLIC_ADDRESS_BASE58_PREFIX = 0x2bb39a;  // Sumo
+  uint64_t const CRYPTONOTE_PUBLIC_ADDRESS_BASE58_PREFIX = 0x2bb39a;  // Ryo
```
Comment (Technohacker) [inline src/cryptonote_config.h]: Editing the comment without updating the prefixes could be problematic. The prefixes are changeable with no effect to existing wallets.

File: src/cryptonote_core/checkpoints.cpp [hunk 2]
```diff
@@ -209,20 +210,20 @@ namespace cryptonote
...
   {
-    // All SumoPulse domains have DNSSEC on and valid
+    // All RyoPulse domains have DNSSEC on and valid
     static const std::vector<std::string> dns_urls = 
     {
-	"checkpoints.sumopulse.stream"
-	, "checkpoints.sumopulse.download"
-	, "checkpoints.sumopulse.win"
-	, "checkpoints.sumopulse.bid"
+      "checkpoints.sumopulse.stream"
```
Comment (Technohacker) [inline src/cryptonote_core/checkpoints.cpp]: We'll need to host our own DNS Pulse servers and rename these

Comment (psychocrypt) [reply src/cryptonote_core/checkpoints.cpp]: Yes must be updated in an separate Pull Request

File: src/wallet/wallet2.cpp [hunk 3]
```diff
@@ -84,8 +85,8 @@ using namespace cryptonote;
 // arbitrary, used to generate different hashes from the same input
 #define CHACHA8_KEY_TAIL 0x8c
 
-#define UNSIGNED_TX_PREFIX "Sumokoin unsigned tx set\002"
-#define SIGNED_TX_PREFIX "Sumokoin signed tx set\002"
```
Comment (fireice-uk) [inline src/wallet/wallet2.cpp]: Those prefixes cannot change yet, without breaking wallet exports.

**Comment by psychocrypt** [comment]
seed server are updated

**Comment by fireice-uk** [comment]
@psychocrypt please address https://github.com/ryo-currency/ryo-emergency/pull/1#discussion_r192420081

**Comment by psychocrypt** [comment]
@fireice_uk what do you mean all comments are adressed. Checkpointings dns
server are disabled. Thats the currently only open comment

**Comment by fireice-uk** [comment]
-#define UNSIGNED_TX_PREFIX "Sumokoin unsigned tx set\002"
-#define SIGNED_TX_PREFIX "Sumokoin signed tx set\002"

Those prefixes need to stay. See the linked comment.

**Comment by fireice-uk** [comment]
Ah, sorry, my bad.

**Pull request merged**